AK/OwnPtrWithCustomDeleter.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022, Lucas Chollet <lucas.chollet@free.fr>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Function.h>
+#include <AK/Noncopyable.h>
+#include <AK/StdLibExtras.h>
+
+template<Pointer T>
+struct OwnPtrWithCustomDeleter {
```

Comment:
Syntactically, it feels weird that this needs a pointer type when all other smart pointers take any type and store a pointer to it. That is, it feels kinda awkward that this is the resulting syntax:
```c++
OwnPtrWithCustomDeleter<addrinfo const*> foo;
OwnPtr<addrinfo> bar;
```

I might have missed some discussion, but is there any reason not to use the same syntax as `OwnPtr`? Then the constructor can accept a `T*`, and so on.

Reply:
I changed it!